src/transactionIntentsFactories/delegationTransactionIntentsFactory.ts

```diff
@@ -0,0 +1,344 @@
+            dataParts = dataParts.concat(options.publicKeys[i].hex());
+            dataParts = dataParts.concat(byteArrayToHex(options.signedMessages[i]));
+        }
+
+        return new TransactionIntentBuilder({
+            config: this.config,
+            sender: options.sender,
+            receiver: options.delegationContract,
+            dataParts: dataParts,
+            executionGasLimit: this.computeExecutionGasLimitForNodesManagement(numNodes)
+        }).build();
+    }
+
+    private computeExecutionGasLimitForNodesManagement(numNodes: number): BigNumber.Value {
+        const additionalGasForAllNodes = new BigNumber(this.config.additionalGasLimitPerValidatorNode).multipliedBy(numNodes);
+        return new BigNumber(this.config.gasLimitDelegationOperations).plus(additionalGasForAllNodes);
+    }
+
+    createTransactionIntentForRemovingNodes(options: {
+        sender: IAddress,
+        delegationContract: IAddress,
+        publicKeys: ValidatorPublicKey[]
+    }): TransactionIntent {
+        let dataParts = ["removeNodes"];
+
+        for (const key of options.publicKeys) {
+            dataParts = dataParts.concat(key.hex());
+        }
+
+        const numNodes = options.publicKeys.length;
+
+        return new TransactionIntentBuilder({
+            config: this.config,
+            sender: options.sender,
+            receiver: options.delegationContract,
+            dataParts: dataParts,
+            executionGasLimit: this.computeExecutionGasLimitForNodesManagement(numNodes)
+        }).build();
+    }
+
+    createTransactionIntentForStakingNodes(options: {
+        sender: IAddress,
+        delegationContract: IAddress,
+        publicKeys: ValidatorPublicKey[]
+    }): TransactionIntent {
+        let dataParts = ["stakeNodes"];
+
+        for (const key of options.publicKeys) {
+            dataParts = dataParts.concat(key.hex());
+        }
+
+        const numNodes = options.publicKeys.length;
+        const executionGasLimit = new BigNumber(numNodes).multipliedBy(
+            this.config.additionalGasLimitPerValidatorNode).plus(
+                this.config.gasLimitStake
+            ).plus(this.config.gasLimitDelegationOperations);
+
+        return new TransactionIntentBuilder({
+            config: this.config,
+            sender: options.sender,
+            receiver: options.delegationContract,
+            dataParts: dataParts,
+            executionGasLimit: executionGasLimit
+        }).build();
+    }
+
+    createTransactionIntentForUnbondingNodes(options: {
+        sender: IAddress,
+        delegationContract: IAddress,
+        publicKeys: ValidatorPublicKey[]
+    }): TransactionIntent {
+        let dataParts = ["unBondNodes"];
+
+        for (const key of options.publicKeys) {
+            dataParts = dataParts.concat(key.hex());
+        }
+
+        const numNodes = options.publicKeys.length;
+        const executionGasLimit = new BigNumber(numNodes).multipliedBy(
```

Comment:
Can be split, a temporary variable can be used.

Reply:
Done